pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -69,12 +69,17 @@ type BuildSpec struct {
 	Notary    *NotaryConfig               `json:"notary,omitempty"`
 }
 
+type BuildCacheConfig struct {
+	VolumeName string `json:"volumeName,omitempty"`
+	ImageTag   string `json:"imageTag,omitempty"`
```

Comment:
In the spirit of reproducibility should we provide an image with a fully qualified digest here?

Reply:
We think no: If "someone" pushed an update to the cache we'd like to always use the most current version.

Reply:
The current [image reference](https://github.com/pivotal/kpack/blob/d5c86c908752e6271eb27179a18ae493705d9007/pkg/apis/build/v1alpha1/build_pod.go#L301-L305) that is provided to the analyze step is the fully qualified built image reference from the last build. This allows deterministic builds that will not run into outside interference if something else is writing to the same tag. I think the same thing applies here, another image could write to the provided cache image tag and we would want to ensure we are using the exact cache digest from the last build.

Reply:
Got it. Should be implemented with 6b3e580df3c99b7ed57dd1ac50730f3da1ed2474 and 177e1397ba8e3112cc9ae28fc2840e529f4c5071. I'll also add some test for that. WDYT?